tests/operators/test_filters.py

```diff
@@ -179,6 +179,24 @@ def test_generate_mask_fail_wrong_condition(cube):
         filters.generate_mask(cube, "!<", 10)
 
 
+def test_generate_mask_rename(cube):
+    """Generates a mask and checks rename."""
+    mask = cube.copy()
+    mask.data[:] = 0
+    mask.data[cube.data == 276] = 1
+    mask.rename(f"mask_for_{cube.name()}_==_276")
+    assert mask.name() == filters.generate_mask(cube, "==", 276).name()
+
+
+def test_generate_mask_units(cube):
+    """Generates a mask and checks units."""
+    mask = cube.copy()
+    mask.data[:] = 0
+    mask.data[cube.data == 276] = 1
+    mask.units = "1"
+    assert mask.units == filters.generate_mask(cube, "==", 276).units
```

Comment:
```suggestion
    expected = cf_units.Unit("1")
    assert filters.generate_mask(cube, "==", 276).units == expected
```
As we are only testing the units there is no need to construct a full cube.